ctapipe/calib/camera/calibrator.py

```diff
@@ -15,62 +15,59 @@
 
 
 def integration_correction(
-    n_chan, pulse_shape, refstep, time_slice, window_width, window_shift
+    reference_pulse_shape, reference_pulse_step, sampled_step, window_width, window_shift
 ):
     """
-    Obtain the integration correction for the window specified.
+    Obtain the correction for the integration window specified.
 
-    This correction accounts for the cherenkov signal that may be missed due
-    to a smaller integration window by looking at the reference pulse shape.
+    For any integration window applied to a noise-less unit pulse, the
+    correction (returned by this function) multiplied by the integration
+    result should equal 1.
 
-    Provides the same result as set_integration_correction from readhess.
+    This correction therefore corrects for the Cherenkov signal that may be
+    outside the integration window, and removes any dependence of the resulting
+    image on the window_width and window_shift parameters. However, the width
+    and shift of the window should still be optimised for the pulse finding and
+    to minimise the noise included in the integration.
 
     Parameters
     ----------
-    n_chan : int
-        Number of gain channels for the telescope
-    pulse_shape : ndarray
-        Numpy array containing the pulse shape for each channel.
-    refstep : int
+    reference_pulse_shape : ndarray
+        Numpy array containing the pulse shape for each gain channel
+    reference_pulse_step : float
         The step in time for each sample of the reference pulse shape
-    time_slice : int
-        The step in time for each sample of the waveforms
+    sampled_step : int
+        The step in time for each sample of the sampled waveforms
```

Comment:
This could be documented better- I'm not sure I understand what it is from the description. It sounds like it should be the width of the time bins for the waveform, but that would not be an integer necessarily.  Does it have a unit like ns?  Maybe calling it sample_width?

Reply:
Thanks, done